src/main/java/bot/assembly/BotBrain.java

```diff
@@ -0,0 +1,105 @@
+package bot.assembly;
+
+import java.util.Scanner;
+
+import bot.assembly.function.BotCommandResponderUnit;
+import bot.assembly.function.BotPrinter;
+import bot.assembly.function.BotTaskStatusGeneratorUnit;
+import bot.assembly.memory.BotDynamicMemoryUnit;
+import bot.assembly.memory.BotStaticMemoryUnit;
+import bot.assembly.memory.CommandInput;
+
+/**
+ * A class that assembles all features of the bot
+ */
+public class BotBrain {
+
+    private BotPrinter botPrinter = new BotPrinter();
+    private BotCommandResponderUnit botCommandResponderUnit = new BotCommandResponderUnit();
+    private BotTaskStatusGeneratorUnit botTaskGeneratorUnit = new BotTaskStatusGeneratorUnit();
+    private BotStaticMemoryUnit botStaticMemoryUnit = new BotStaticMemoryUnit();
+    private BotDynamicMemoryUnit botDynamicMemoryUnit = BotDynamicMemoryUnit.getInstance();
+
+    private boolean isTerminated = false;
+
+    /**
+     * Constructor
+     */
+    public BotBrain() {}
+
+    /**
+     * A method that receives the command input and differentiate the type of the command
+     * using the Enum class CommandInput. Then proceed to call the respective actions using
+     * various units from different packages
+     * @param input
+     * @throws Exception
+     */
+    private void reactToCommand(String input) throws Exception {
+
+        CommandInput commandInitial = botCommandResponderUnit.identifyCommand(input);
+
+        switch (commandInitial) {
+        case BYE:
+            botPrinter.print(botStaticMemoryUnit.MESSAGE_GOODBYE);
+            isTerminated = true;
+            return;
+        case LIST:
+            botTaskGeneratorUnit.generateTaskTrackerReport();
+            break;
+        case DONE:
+            botCommandResponderUnit.markTaskAsDone(input);
+            break;
+        case DELETE:
+            botCommandResponderUnit.deleteTaskFromList(input);
+            break;
+        case FIND:
+            botCommandResponderUnit.findTaskFromList(input);
+            break;
+        default:
+            botCommandResponderUnit.addTask(input);
+            botTaskGeneratorUnit.generateAddTaskFeedback();
+        }
+    }
+
+    /**
+     * A method that attempts to initiate the loading process of data from HARD_DISK
+     */
+    private void wakeUpMemory() {
+
+        try {
+            botDynamicMemoryUnit.loadFromHardDisk();
+            botTaskGeneratorUnit.generateTaskTrackerReport();
+        } catch (Exception error) {
+            botPrinter.print(botStaticMemoryUnit.ERROR_MESSAGE_PROMPT + error.getMessage());
+        }
+
+    }
+
+    /**
+     * A method that initiate reaction of Bot to user
+     */
+    private void interact() {
```

Comment:
This seems to be a grey area between the bot and the UI, maybe more modular code can help make things clearer?